src/components/Uploader/Uploader.tsx

```diff
@@ -0,0 +1,214 @@
+      maxSize = 1048576,
+      name = "files",
+      files = [],
+      onChange,
+      children,
+      ...restProps
+    },
+    ref
+  ) => {
+    const [isPreviewOpen, setIsPreviewOpen] = useState(false)
+    const {
+      isDragActive,
+      isDragReject,
+      getRootProps,
+      getInputProps,
+    } = useDropzone({
+      accept,
+      multiple,
+      maxSize,
+      ...restProps,
+
+      onDrop: (acceptedFiles) => {
+        const newFiles = multiple ? files.concat(acceptedFiles) : acceptedFiles
+
+        onChange?.(newFiles)
+      },
+    })
+
+    function handleFileRemove(
+      event: React.MouseEvent<HTMLButtonElement>
+    ): void {
+      const fileIndex = event.currentTarget.dataset.index
+      if (!fileIndex) return
+
+      const newFiles = files.filter(
+        (file, index) => index !== Number(fileIndex)
+      )
+      onChange?.(newFiles)
+    }
+
+    return (
+      <div
+        sx={{
+          display: "flex",
+          flexDirection: "column",
+          alignItems: "flex-start",
+          maxWidth: "460px",
+          textAlign: "center",
+        }}
+      >
+        <div
+          sx={{
+            paddingY: 4,
+            paddingX: 6,
+            border: "1px dashed",
+            borderColor: "secondary.40",
+            borderRadius: " 12px",
+            cursor: "pointer",
+            transition: "all .2s linear",
+
+            ...stateStyles[isDragActive ? "hover" : "resting"],
+
+            "&:hover": {
+              ...stateStyles.hover,
+            },
+          }}
+          {...getRootProps({ className: "dropzone" })}
+        >
+          <input
+            type="file"
+            name={name}
+            ref={ref}
+            title="Drag & Drop or browse a file"
+            {...getInputProps()}
+          />
+          {children}
+        </div>
+
+        {isDragReject && <Text>File type not accepted, sorry!</Text>}
+
+        {files.length > 0 && (
+          <Flex
+            sx={{
+              justifyContent: "space-between",
+              alignItems: "center",
+              alignSelf: "stretch",
+              marginTop: 3,
+              marginBottom: 2,
+            }}
+          >
+            <Text
+              sx={{
+                color: "text.40",
+                fontSize: 13,
+              }}
+            >
+              {files.length} files selected
+            </Text>
+            <Button
+              variant="plain"
+              color="secondary"
+              onClick={() => setIsPreviewOpen((previous) => !previous)}
+            >
+              {!isPreviewOpen ? "View & manage files" : "Hide files"}
+            </Button>
+          </Flex>
+        )}
+
+        <div
+          sx={{
+            display: "grid",
+            opacity: isPreviewOpen ? 1 : 0,
+            visibility: isPreviewOpen ? "visible" : "hidden",
+            maxHeight: isPreviewOpen ? "9999px" : 0,
+            gridTemplateColumns: "50% 50%",
```

Comment:
Avoid using percentages for this as it refers to the total container width, and not the remaining width after the `gaps` are distributed, so you end up with columns overflowing the container like so:
![image](https://user-images.githubusercontent.com/12552164/111796913-6c21c300-88a7-11eb-94a7-774812c6bdc5.png)

```suggestion
            gridTemplateColumns: "1fr 1fr",
```